### PR TITLE
IBX-497: As an Editor, I want to see redesigned tabs

### PR DESCRIPTION
**Pull request by tischsoic**
| Question      | Answer
| ------------- | ---
| Tickets       | https://issues.ibexa.co/browse/IBX-497
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Tests pass?   | yes
| Doc needed?   | ???
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

# Related PRs:
- https://github.com/ezsystems/ezplatform-icons/pull/5
- https://github.com/ezsystems/ez-support-tools/pull/98
- https://github.com/ezsystems/ezcommerce-admin-ui/pull/91
- https://github.com/ezsystems/ezcommerce-shop/pull/314
- https://github.com/ezsystems/ezplatform-workflow/pull/194
- https://github.com/ezsystems/ezplatform-personalization/pull/210
- https://github.com/ezsystems/ezplatform-connector-dam/pull/34
- https://github.com/ezsystems/ezplatform-page-builder/pull/770
- https://github.com/ezsystems/ezplatform-form-builder/pull/285
- https://github.com/ezsystems/ezplatform-segmentation/pull/29

# New tabs design
Screenshot 2021-06-21 at 11 14 17

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/bundle/Resources/views/themes/admin/content/location_view.html.twig [hunk 1]
```diff
@@ -76,6 +76,7 @@
             <div class="panel panel-primary">
                 <div class="panel-body">
                     {# 'is_location_visible': location.invisible - param deprecated since EZP-32395 #}
+                    asdf
```
Comment (GrabowskiM) [inline src/bundle/Resources/views/themes/admin/content/location_view.html.twig:79]: There's asdf in this line.

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

File: src/bundle/Resources/public/js/scripts/admin.location.adaptive.tabs.js [hunk 2]
```diff
@@ -1,147 +1,148 @@
...
+        toggler.classList.toggle('ibexa-tabs__toggler--rolled-up');
+        tabContent.style.height = isTabContentRolledUp ? 'auto' : '0px';
     };
 
     copyTabs();
-    adaptTabs();
 
-    doc.addEventListener('click', handleClickOutsideSecondaryMenu);
+    doc.querySelectorAll('.ibexa-tabs__toggler').forEach((toggler) => {
+        toggler.addEventListener('click', toggleContainer);
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/admin.location.adaptive.tabs.js:72]:
```suggestion
        toggler.addEventListener('click', toggleContainer, false);
```

File: src/bundle/Resources/public/js/scripts/admin.location.adaptive.tabs.js [hunk 3]
```diff
@@ -1,147 +1,148 @@
...
+    });
+
+    doc.addEventListener('click', handleClickOutsidePopupMenu);
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/admin.location.adaptive.tabs.js:75]:
```suggestion
    doc.addEventListener('click', handleClickOutsidePopupMenu, false);
```

File: src/bundle/Resources/public/js/scripts/admin.location.adaptive.tabs.js [hunk 4]
```diff
@@ -1,147 +1,148 @@
...
+                    adaptiveItems.adapt();
                 });
             });
         }
     });
 
-    document.body.addEventListener('ez-content-tree-resized', adaptTabs);
-    window.addEventListener('resize', adaptTabs);
-})(window, window.document, window.jQuery, window.Translator);
+    document.body.addEventListener('ez-content-tree-resized', adaptAlltabs);
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/admin.location.adaptive.tabs.js:146]:
```suggestion
    document.body.addEventListener('ez-content-tree-resized', adaptAlltabs, false);
```

File: src/bundle/Resources/views/themes/admin/ui/component/popup_menu/popup_menu.html.twig [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<ul class="ibexa-popup-menu {{ class|default('') }}">
+    {% for item in items %}
+        {{ include('@ezdesign/ui/component/popup_menu/popup_menu_item.html.twig', {
```
Comment (GrabowskiM) [inline src/bundle/Resources/views/themes/admin/ui/component/popup_menu/popup_menu.html.twig:3]: I think it's bad indent below?

File: src/bundle/Resources/views/themes/admin/ui/tab/location_view.html.twig [hunk 6]
```diff
@@ -1,24 +1,49 @@
 <div class="ez-header">
     <div class="container">
-        <ul class="nav nav-tabs ez-tabs" role="tablist" id="ez-tab-list-{{ group }}">
+    <div class="ibexa-tabs">
```
Comment (GrabowskiM) [inline src/bundle/Resources/views/themes/admin/ui/tab/location_view.html.twig:3]: bad indent

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

File: src/bundle/Resources/views/themes/admin/ui/component/popup_menu/popup_menu.html.twig (hunk 5 above)
Comment (tischsoic) [reply src/bundle/Resources/views/themes/admin/ui/component/popup_menu/popup_menu.html.twig:3]: Maybe?

File: src/bundle/Resources/public/js/scripts/admin.content.tree.js [hunk 7]
```diff
@@ -71,6 +71,28 @@
...
+    };
+    const handleContainerTransitionStop = () => {
+        transitionCount -= 1;
+
+        if (transitionCount == 0) {
+            clearInterval(transitionEventIntervalId);
+            dispatchContentTreeResizeEvent();
+        }
+    };
+    contentTreeContainer.addEventListener('transitionstart', handleContainerTransitionStart, false);
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/admin.content.tree.js:92]:
```suggestion

    contentTreeContainer.addEventListener('transitionstart', handleContainerTransitionStart, false);
```

File: src/bundle/Resources/public/js/scripts/admin.location.adaptive.tabs.js [hunk 8]
```diff
@@ -1,147 +1,184 @@
...
     };
-    const handleClickOutsideSecondaryMenu = (event) => {
-        doc.querySelectorAll(TABS_SELECTOR).forEach((primaryTabsList) => {
-            const moreTab = primaryTabsList.querySelector('.ez-tabs__tab--more');
+    const handleTabsConainterChange = () => {
+        if (frame) {
+            cancelAnimationFrame(frame);
+        }
+
+        frame = requestAnimationFrame(adaptTabsAndPopupMenu);
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/admin.location.adaptive.tabs.js:65]: I wonder if frame is correct name, maybe sth like animationFrame? As frame is quite vast term

File: src/bundle/Resources/public/js/scripts/core/adaptive.items.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    const OFFSET_ROUNDING_COMPENSATOR = 0.5;
+
+    class AdaptiveItems {
+        constructor(config) {
+            this.items = config.items;
+            this.selectorItem = config.selectorItem;
+            this.itemHiddenClass = config.itemHiddenClass;
+            this.container = config.container;
+            this.getActiveItem = config.getActiveItem;
+
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/core/adaptive.items.js:11]: why empty line?

File: src/bundle/Resources/public/js/scripts/core/adaptive.items.js [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+            for (let i = 0; i < this.items.length; i++) {
+                const item = this.items[i];
+
+                if (item === activeItem) {
+                    continue;
+                }
+
+                const lastItem = this.items[this.items.length - 1]
+                const isLastNonactiveItem = lastItem === activeItem ? i === this.items.length - 2 : i === this.items.length - 1;
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/core/adaptive.items.js:33]: what would you say for `this.items.length - (lastItem === activeItem ? 2 : 1)`?

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

File: src/bundle/Resources/public/js/scripts/core/adaptive.items.js (hunk 10 above)
Comment (tischsoic) [reply src/bundle/Resources/public/js/scripts/core/adaptive.items.js:33]: So, you are proposing this?
```suggestion
                const isLastNonactiveItem = i === this.items.length - (lastItem === activeItem ? 2 : 1);
```
It is shorter, but for me my version is easier to understand. 
Do you think that your version is easier to understand?

Comment (GrabowskiM) [reply src/bundle/Resources/public/js/scripts/core/adaptive.items.js:32]: I don't. ;) But also I don't like using comparison inside ternary operator, now you have three `===` in one line

**Review by dew326 — APPROVED** [review]

**Review by lucasOsti — APPROVED** [review]

**Pull request merged**